### PR TITLE
Implement health check interval user setting

### PR DESCRIPTION
**Pull request by Steve-Mcl**
## Description

* Adds new key to instance settings `KEY_HEALTH_CHECK_INTERVAL`
* Updates API to permit `launcherSettings` to be passed in/out
* Adds new page `instance/<instance-id>/settings/launcher` 

NOTES:
* Can be approved any time, but should only be merged once all related PRs are approved:
   * https://github.com/FlowFuse/driver-localfs/pull/124
   * https://github.com/FlowFuse/driver-docker/pull/92
   * https://github.com/FlowFuse/driver-k8s/pull/150
   * https://github.com/FlowFuse/nr-launcher/pull/228
* Does not create yet another API, instead utilises the existing "Update a project" route PUT `/api/v1/projects/:id`
* Updates the SCHEMA to pass/present the new setting in an object `launcherSettings` (negates need for future changes to API schema)
* Stores value in `ProjectSettings` under own KEY `KEY_HEALTH_CHECK_INTERVAL`
   * This could be changed now (or in the future) to be a sub property of an `object` if desired (would not affect public facing API since that was set up as an object by design for such eventualities)

#### Settings:
image

#### In actions:
image

#### Audit log entry:
image

### TODO
- [x] tests

## Related Issue(s)

#3715

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/pages/instance/Settings/LauncherSettings.vue
Comment (knolleary) [inline frontend/src/pages/instance/Settings/LauncherSettings.vue:9]:
```suggestion
            Health check interval (ms)
            <template #description>
                The interval at which the launcher will check the health of Node-RED. Flows that perform CPU intensive work may require to increase this from the default of 7500ms.
                It is not recommended to set this value lower than 5000ms.
            </template>
```

File: forge/routes/api/project.js
Comment (knolleary) [inline forge/routes/api/project.js:487]: The UI text says a value below 5000 is not recommended - I'd suggest we apply that as a minimum value here as well, and add some UI feedback that prevents a smaller value from being submitted.

**Review by knolleary — CHANGES_REQUESTED** [review]
A couple suggested changes.

I'm iffy on the idea of adding a Launcher section to Instance settings as it becomes another concept for a user to have to understand the meaning of. That said, I don't think we have quite a suitable existing location for such a setting. So won't block for that - but the other bits I would like looked at.

File: forge/routes/api/project.js
Comment (Steve-Mcl) [reply forge/routes/api/project.js:487]: Yes, I did set an absolute minimum (safe guard), the other number was a recommendation. 

Will apply 5s as the hard minimum though.

File: test/unit/forge/routes/api/project_spec.js
Comment (Steve-Mcl) [inline test/unit/forge/routes/api/project_spec.js:1701]:
```suggestion
                        healthCheckInterval: 999 // 999 is below the 5000 is the minimum
```

File: forge/routes/api/project.js
Comment (Steve-Mcl) [reply forge/routes/api/project.js:487]: limit applied in [89db3d2](https://github.com/FlowFuse/flowfuse/pull/3716/commits/89db3d2acc9888145cb05d07b0d4e748ac899b55)

checking alert suitability (will adjust if not suitable)

Comment (Steve-Mcl) [reply forge/routes/api/project.js:487]: @knolleary Added specific alert (commit: [9412788](https://github.com/FlowFuse/flowfuse/pull/3716/commits/9412788ca8b7846b411376262ed2276b3ac37a3f) ) but it kinda makes part of the description field (highlighted in yellow) a bit redundant 

image

Comment (knolleary) [reply forge/routes/api/project.js:487]: I was more thinking about validation before the user can click save as we do elsewhere, for example, the username field:

image

Comment (Steve-Mcl) [reply forge/routes/api/project.js:487]: Added client side validation 

image


commit incomming.

Comment (Steve-Mcl) [reply forge/routes/api/project.js:487]: Added in [8378317](https://github.com/FlowFuse/flowfuse/pull/3716/commits/8378317c63bd7fa6d78b4a18e0997556f9b9cdf8)

Also adds e2e test in [81588d2](https://github.com/FlowFuse/flowfuse/pull/3716/commits/81588d21c592b3145107aee4c7bf4c4f570f6306)

```
FlowFuse - Instance - Settings - Launcher
    Validates health check interval user input
```

**Comment by Steve-Mcl** [comment]
@knolleary, hold fire on review/merge please. Not able to navigate to new tab in pre-staging env tests.

**Comment by Steve-Mcl** [comment]
@knolleary 

super super wired issue on pre-staging where i could not navigate to the new page. I could not recreate on local.
pushed a simple change and now it works! No idea why.

Then followed an odd PG test fail
> compares relative to now if only a to date is passed
> `expect(elapsedTime(soon)).toBe('30 minutes, 15 seconds')`

value was `30 minutes, 14 seconds` - some kind of flaky timing? re-run and ok!

Hey ho. converting back to "ready" now we have all greens.

Tested on pre-staging OK.

**Review by knolleary — APPROVED** [review]

**Comment by joepavitt** [comment]
Anything blocking this from moving forward? Been sat for 2 weeks

**Comment by Steve-Mcl** [comment]
> Anything blocking this from moving forward? Been sat for 2 weeks

Not as far as I am concerned Joe. Possibly the far reaching nature (5x PRs), the timing of their individual approval/branch update/tests-reruns etc was the reason for approval without merge.

**Comment by joepavitt** [comment]
Do we need to have the other PRs merged/released before pushing this into production?

**Comment by knolleary** [comment]
I'm taking a look at this as I don't quite see how the setting is getting into the launcher from the k8s/docker driver changes.

Please hold off any merging until I update on that.

In terms of going live - as this has a launcher change, that will require stack updates in production after publishing... that is not something we automate today

**Review by knolleary — CHANGES_REQUESTED** [review]
I've taken a second look at this and I don't see any need to update all the drivers. We can centralise it in the /settings api endpoint in the forge app - https://github.com/FlowFuse/flowfuse/blob/main/forge/routes/api/project.js#L798 - which takes the settings from the driver and adds in a bunch of standard settings (which this would be one).

The driver settings are intended to be the driver-specific things that need setting differently depending on the needs of the driver.

**Comment by Steve-Mcl** [comment]
Added `healthCheckInterval` to settings in the `/settings` api endpoint in the forge app https://github.com/FlowFuse/flowfuse/blob/main/forge/routes/api/project.js#L798 as suggested in the linked comment.

<BR>

NOTE:

The PG test fail is unrelated and was [raised as a chore](https://github.com/FlowFuse/flowfuse/issues/3735) - found when implementing SQL Server as backend DB. It has finally bitten!


### Test fail: 
```
 1) Device API
       Device Checkin
         device checks in with a valid snapshot and settingsHash:

      AssertionError: expected 409 to be 200
      + expected - actual

      -409
      +200
      
      at Assertion.fail (node_modules/should/cjs/should.js:275:17)
      at Assertion.value (node_modules/should/cjs/should.js:356:19)
      at Context.<anonymous> (test/unit/forge/routes/api/device_spec.js:1738:40)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

failing due to missing `await` on `dbDevice.updateSettings({ settings: { env: [{ name: 'FOO', value: 'BAR' }] } })` here: https://github.com/FlowFuse/flowfuse/blob/3c5e0276eb3c96511378571e1d8f3009090d2b27/test/unit/forge/routes/api/device_spec.js#L1715


I will raise a PR and once merged, re-run tests.

**Comment by Steve-Mcl** [comment]
Rerun tests once [Fix async and transactional issues in code base #3812](https://github.com/FlowFuse/flowfuse/pull/3812) is merged

**Review by knolleary — APPROVED** [review]

**Comment by knolleary** [comment]
My only pause on merging this right now is that it depends on the updated `nr-launcher` to pass through the setting; which requires updated stack builds. We don't have the CI/CD process for that cracked yet. That said, we're doing the release tomorrow which will update the stacks. I think it is low risk to merge this now, but hold off highlighting it until the release is done tomorrow and the stacks available.

**Pull request merged**
